docs/source/reference/api-server/api-server-upgrade.rst

```diff
@@ -0,0 +1,218 @@
+    
+    After the patch, verify the API server is cordoned again.
+
+3. Drain the old API server by waiting or canceling current requests:
+
+.. tab-set::
+
+    .. tab-item:: Inspecting requests
+
+        You can inspect the status of requests by running:
+
+        .. code-block:: console
+
+            $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl exec {} -c skypilot-api -- sky api status
+            sky api status
+            ID                                    User             Name        Created         Status
+            942f6ab3-f5b6-4a50-acd6-0e8ad64a3ec2  <USER>           sky.launch  a few secs ago  PENDING
+            8c5f19ca-513c-4068-b9c9-d4b7728f46fb  <USER>           sky.logs    26 secs ago     RUNNING
+            skypilot-status-refresh-daemon        skypilot-system  sky.status  25 mins ago     RUNNING
+
+        .. note::
+
+            The `skypilot-status-refresh-daemon` is a background process managed by API server that can be safely interrupted.
+    
+    .. tab-item:: Canceling requests
+
+        You can cancel less critical requests by running:
+
+        .. code-block:: console
+
+            $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl exec {} -c skypilot-api -- sky api cancel ${ID}
+
+.. dropdown:: Using port-forward to access the API server
+
+    If you do not have `exec` access to the API server Pod, you can also use `port-forward` to access the api status:
+
+    .. code-block:: console
+
+        $ kubectl get po -l app=${RELEASE_NAME}-api -oname | xargs -I {} kubectl port-forward {} 46580:46580 > /tmp/port-forward.log 2>&1 &
+        $ PORT_FORWARD_PID=$!
+        $ sky api login -e http://127.0.0.1:46580
+        # Polling the status
+        $ sky api status
+        # Cancel less critical requests if needed
+        $ sky api cancel ${ID}
+        # Stop the port-forward after you are satisfied with the status
+        $ kill $PORT_FORWARD_PID
+
+Step 3: Upgrade SkyPilot clients
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, compatibility between SkyPilot clients and server is only guaranteed when both sides are running on **the same version**. We recommend upgrading all the clients to the same version during the maintainence window to avoid breaking the pipeline after the upgrade:
+
+.. code-block:: bash
+
+    pip install -U skypilot-nightly==${VERSION}
+
+Step 4: Upgrade the API server
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Once all the critical requests have been finished and the clients in autonomous pipelines have been upgraded, you can upgrade the API server with the following command:
+
+.. code-block:: bash
+
+    # --reuse-values is critical to keep the values set in the previous installation steps.
+    helm upgrade -n $NAMESPACE $RELEASE_NAME skypilot/skypilot-nightly --devel --reuse-values \
+      --set apiService.image=${IMAGE_REPO}:${VERSION}
```

Comment:
IIRC, we had a `--version` argument, can it be used?

https://docs.skypilot.co/en/latest/reference/api-server/api-server-admin-deploy.html#step-3-deploy-the-api-server-helm-chart

Can we directly use the default image repo specifed in our values.yaml, so people don't need to manually set the `IMAGE_REPO` env var?

Reply:
`--version` specifies the version of our **helm chart**. For the repo thing, I lean to use `.image` for simplicity, there are charts (like nginx-ingress) do separate image to `repo` and `tag` config. Here's a comparison:

Pros of sticking in `.image`:

- The final image to use and the way to customize repository are straightforward
- Better interoperate with pipeline: just use the image built in previous steps without parsing
- No compatibility issue to handle for us

Cons:

- Exposing more details to inexperienced users
- More verbose, such as this doc

I think in general the difference is trivial compared to our main functionality, how about we just stick in the `.image` method?

Reply:
Good point! Thanks for the detailed explanation. `.image` sounds good to me.